aodncore/pipeline/steps/resolve.py

```diff
@@ -318,6 +338,52 @@ def run(self):
         return self._collection
 
 
+class DeleteManifestResolveRunner(BaseManifestResolveRunner):
+    """Handles a delete manifest file which only contains a list of source files, and a valid "deletion publish type"
+        string, as defined by the PipelineFilePublishType enum
+
+    File format must be as follows::
+
+        destination/path/for/delete1,DELETE_PUBLISH_TYPE
+        destination/path/for/delete2,DELETE_PUBLISH_TYPE
```

Comment:
I hadn't thought of specifying the publish type in the manifest. I guess this could be useful, but I think it most cases it would be up to the specific handler to determine what it needs to do. Could we allow this to be left blank, in which case the handler decides?

Reply:
Yeah that's a fair point. I was thinking in terms of the *person* deciding when they create the manifest but you make a good point about the handler logic being able to decide on a file by file basis (as it often does for additions!).

Reply:
@mhidas I've addressed this by removing the `required` constraint and not setting the `publish_type` property unless it has been provided for any given row.

https://github.com/aodn/python-aodncore/pull/215/files#diff-dc09111dc00c10cc36c7d376aa0a09095f8aa1503b66cba58d7d31c4e4e79820R2

The only slightly quirky thing with having this optional is that a trailing comma is **required** if the publish type field is omitted, otherwise the row is parsed as only having 1 field instead of 2. This gives the most control/flexibility to people in setting the publish type, with the only caveat being that minimal delete manifests look like this:

```csv
PATH/TO/FILE/1,
PATH/TO/FILE/2,
PATH/TO/FILE/3,
```

So we really have two options which I'm happy to go with either, depending on the usability from your side:

- Option A: if the commas are annoying, remove the publish_type field from the schema and have a single field "CSV" (but the handler will *always* be solely responsible for handling the publish type)
- Option B: leave this feature, and POs creating manifests need to remember the trailing comma (if omitted, the file will land in error and the schema error will be in the process log)

Reply:
@aodn/data-team Any thoughts?

Mine are:
* So far the old `po_s3_del` command has satisfied all my deletion needs (equivalent to `DELETE_UNHARVEST` publish_type, and silently ignoring cases where there's no harvester to unharvest a file), so Option A would pr

Reply:
YAGNI principle would dictate that we retain the current behaviour unless the feature is demonstrated to be useful to anyone. Would be preferable to adding complexity for a feature nobody uses!

Reply:
I think a valid and sane csv do not have trailing commas, unless you want to specify a empty column.

Maybe Option D: Just use a "single file per line" file!? Simple is better. This removes both the "another csvish formatted file to deal with" and the yagni/fields that I don't care problems.

I say that because creating these manifests is usually done via regexes/grep/cut pies or sql where queries. Thus, better to stick with a list of files only which is easily dumped straight away. If the need ever arises for specifying more than one thing for each , then we are better of creating a proper json/yaml manifest. 

Otherwise, just allow any csv input and pickup the columns that matches the columns on the database ('file_url",'publish/harvest_type',etc). At least this way we don't need reformatting and we can just feed in directly from a SQL query dump.

Reply:
If I'm reading all this correctly, what @lwgordonimos means by "retain the current behaviour" and what @ocehugo means by "single file per line" are one and the same, and this is "Option A" [above](https://github.com/aodn/python-aodncore/pull/215#discussion_r508946484).